rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchDomain.scala

```diff
@@ -0,0 +1,307 @@
+                                                 override val attributes : Set[QSAttribute] = Set(RuleId, DirectiveIds, Targets)
+  }
+
+  final val all: Set[QSObject] = sealerate.values[QSObject]
+
+  // default sort for QuickSearchResult:
+  // - by type
+  // - then by name
+  def sortQSObject(a: QSObject, b:QSObject): Boolean = {
+
+    implicit class QSObjectOrder(o: QSObject) {
+      def order() = o match {
+        case Common    => 0
+        case Node      => 1
+        case Group     => 2
+        case Directive => 3
+        case Parameter => 4
+        case Rule      => 5
+      }
+    }
+    a.order <= b.order
+  }
+}
+
+/**
+ * Mapping between human and domain name.
+ * Mostly used in the query parser, but also in the
+ * documentation.
+ */
+final object QSMapping {
+  /**
+   * Mapping between a string and actual objects.
+   * We try to be kind with users: not case sensitive, not plural sensitive
+   */
+  val objectNameMapping = {
+    QSObject.all.map { obj =>
+      val n = obj.name.toLowerCase
+      (n -> obj) :: ( n + "s" -> obj) :: Nil
+    }.flatten.toMap
+  }
+
+  //set of names by attribute
+  val attributeNames = {
+    import QSAttribute._
+    val descriptions = Set(Description, LongDescription).map( _.name ) ++ Set("descriptions")
+
+    QSAttribute.all.map { a => a match {
+      case Name              => (a, Set(Name.name, "name") )
+      case Description       => (a, descriptions)
+      case LongDescription   => (a, descriptions)
+      case IsEnabled         => (a, Set(IsEnabled.name  ))
+      case NodeId            => (a, Set(NodeId.name, "nodeid") )
+      case Fqdn              => (a, Set(Fqdn.name, "fqdn") )
+      case OsType            => (a, Set(OsType.name, "ostype", "os"))
+      case OsName            => (a, Set(OsName.name, "osname", "os") ) //not also full name because osFullname contains osName, not the reverse
+      case OsVersion         => (a, Set(OsVersion.name, "osversion", "os") )
+      case OsFullName        => (a, Set(OsFullName.name, "osfullname", OsName.name, "osname", "os") )
+      case OsKernelVersion   => (a, Set(OsKernelVersion.name, "oskernelversion", "oskernel", "kernel", "os") )
+      case OsServicePack     => (a, Set(OsServicePack.name, "osservicepack", "ossp", "sp", "servicepack", "os") )
+      case Arch              => (a, Set(Arch.name, "arch") )
+      case Ram               => (a, Set(Ram.name, "memory") )
+      case IpAddresses       => (a, Set(IpAddresses.name, "ip", "ips", "networkips") )
+      case PolicyServerId    => (a, Set(PolicyServerId.name, "policyserver") )
+      case Properties        => (a, Set(Properties.name, "node.props", "nodeprops", "node_properties", "nodeproperties") )
+      case RudderRoles       => (a, Set(RudderRoles.name, "serverrole", "serverroles", "role", "roles") )
+      case GroupId           => (a, Set(GroupId.name, "groupid") )
+      case IsDynamic         => (a, Set(IsDynamic.name) )
```

Comment:
Static = not dynamic :)

It would be helpful to have a search operator directly for that though, true.